### PR TITLE
Closes #11819:Remove nativeMessaging permission from the add-on BLOCKED_PERMISSIONS list

### LINKED ISSUE #11819: Remove nativeMessaging permission from the add-on BLOCKED_PERMISSIONS list
Related https://github.com/mozilla-mobile/fenix/issues/15904#issuecomment-1059541289

### PR DESCRIPTION
**Pull request by Amejia481**
### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/main/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: docs/changelog.md [hunk 1]
```diff
@@ -11,6 +11,9 @@ permalink: /changelog/
 * [Gecko](https://github.com/mozilla-mobile/android-components/blob/main/buildSrc/src/main/java/Gecko.kt)
 * [Configuration](https://github.com/mozilla-mobile/android-components/blob/main/.config.yml)
 
+* **feature-addons**
+  * 🚒 Bug fixed [issue #11819] https://github.com/mozilla-mobile/android-components/issues/11819) - Remove nativeMessaging permission from the add-on BLOCKED_PERMISSIONS list.
```
Comment (Rob--W) [inline docs/changelog.md:15]: A more accurate/meaningful way of describing the changes is:

> Do not block installation of extensions with the "nativeMessaging" permission any longer, but ignore the permission instead for non-builtin add-ons.

Comment (Amejia481) [reply docs/changelog.md:15]: 👍🏽

Comment (Amejia481) [reply docs/changelog.md]: Updated!

**Review by Rob--W — APPROVED** [review]

**Comment by Amejia481** [comment]
@Rob--W we are really close to release, do you know the priority/risk on this patch? If it's not something urgent, I would rather wait for landing until the start the new nightly cycle on Tuesday.

**Comment by Rob--W** [comment]
Little risk, but it is not an issue to wait until the version bump.

**Review by csadilek — APPROVED** [review]
In https://github.com/mozilla-mobile/fenix/issues/15904#issuecomment-961995577, @agi mentioned that the "front-end should never get the request for nativeMessaging to begin with. Otherwise they would have to know to not ask the user for permission for nativeMessaging when geckoviewAddons is not present which I think is not the right place to handle that.... I guess worst case scenario we can filter the permission request in GeckoView right before it gets to AC."

I would have preferred that, as it would allow us to keep the explicit check as a fallback for regressions of the underlying fix that makes sure `nativeMessaging` only works with `geckoViewAddons`. OK to me to land this though. @agi can you review as well?

**Review by agi — CHANGES_REQUESTED** [review]
Yeah this is not the right way to approach this, we should not even request the permission if the addon is not a built-in one. Please file a GeckoView bug and we can discuss it there how to do it.

**Comment by Amejia481** [comment]
> Yeah this is not the right way to approach this, we should not even request the permission if the addon is not a built-in one. Please file a GeckoView bug and we can discuss it there how to do it.

@agi I created the bug https://bugzilla.mozilla.org/show_bug.cgi?id=1759601

**Pull request closed without merging**